### PR TITLE
chore: add a span attr to indicate what ListObjects handler was used

### PR DESCRIPTION
**Pull request by jon-whit**
## Description
This PR adds a span attribute to the `ListObjects` span that indicates what handler was used (optimized or non-optimized variant).

## References
This work is essentially superseded by https://github.com/openfga/openfga/pull/522, because at that point it will be obvious which path ListObjects took.

## Review Checklist
- [ ] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [ ] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: pkg/server/commands/list_objects.go [hunk 1]
```diff
@@ -114,6 +119,8 @@ func (q *ListObjectsQuery) handler(
 		// ConnectedObjects currently only supports models that do not include intersection and exclusion,
 		// and the model must include type info for ConnectedObjects to work.
 		if !containsIntersection && !containsExclusion && hasTypeInfo {
+			span.SetAttributes(attribute.Bool("list-objects-optimized", true))
```
Comment (miparnisari) [inline pkg/server/commands/list_objects.go:122]: the previous `set` is inside `handler`... should we do the same here and put it in line 150?

Comment (jon-whit) [reply pkg/server/commands/list_objects.go:122]: Yeah, good call 👍

Comment (jon-whit) [reply pkg/server/commands/list_objects.go]: Ok, moved.

File: pkg/server/commands/list_objects.go [hunk 2]
```diff
@@ -90,6 +93,8 @@ func (q *ListObjectsQuery) handler(
 	}
 
 	handler := func() {
+		span.SetAttributes(attribute.Bool("list-objects-optimized", false))
```
[the remark this answers is not in the document]
Comment (jon-whit) [reply pkg/server/commands/list_objects.go:96]: Ah, yes. Thank you.

**Review by miparnisari — APPROVED** [review]

**Review by miparnisari — APPROVED** [review]

**Pull request merged**